crates/volta-core/src/tool/yarn/resolve.rs

```diff
@@ -109,7 +128,41 @@ fn resolve_latest_legacy(url: String) -> Fallible<Version> {
 }
 
 fn resolve_semver_from_registry(matching: VersionReq) -> Fallible<Version> {
-    let (url, index) = fetch_yarn_index()?;
+    if env::var_os("VOLTA_FEATURE_YARN_3").is_some() {
+        // first try yarn2+, which uses "@yarnpkg/cli-dist" instead of "yarn"
+        let (url, index) = fetch_yarn_index("@yarnpkg/cli-dist")?;
+        let matching_entries: Vec<PackageDetails> = index
+            .entries
+            .into_iter()
+            .filter(|PackageDetails { version, .. }| matching.matches(version))
+            .collect();
+
+        if matching_entries.len() > 0 {
```

Comment:
As suggested by Clippy:
```suggestion
        if !matching_entries.is_empty() {
```